File: CS.NET/PdfViewerCSharpAPI_R2/Utilities/PdfViewerException.cs
/****************************************************************************
 *
 * File:            PdfViewerException.cs
 *
 * Description:     The 3-Heights™  Pdf Viewer Control for WPF
 *
 * Copyright:       Copyright (C) 2015 PDF Tools AG, Switzerland
 *                  All rights reserved.
 *                  
 ***************************************************************************/

namespace PdfTools.PdfViewerCSharpAPI.Utilities
{
    using System;
    using System.Collections.Generic;
    using System.Linq;
    using System.Text;

    /// <summary>
    /// Exception which passes information about issues opening and reading a pdf.
    /// </summary>
    public class PdfViewerException : Exception
    {
        public PdfViewerException() : base()
        {

        }

        public PdfViewerException(string message) : base(message)
        { }
    }

    public class PdfNoFileOpenedException : PdfViewerException
    {
        public PdfNoFileOpenedException() : base("No file is open to process this operation") { }
    }

    public class PdfFileNotFoundException : PdfViewerException
    {
        public PdfFileNotFoundException()
            : base("File was not found")
        { }
        public PdfFileNotFoundException(string message)
            : base(message)
        { }
    }

    public class PdfFileCorruptException : PdfViewerException
    {
        public PdfFileCorruptException(string message) : base(message) { }
    }
    public class PdfPageNotCachedException : PdfViewerException
    {
        public PdfPageNotCachedException(IList<int> missingPages)
        {
            this.missingPages = missingPages;
        }

        public IList<int> missingPages;
    }
    public class PdfLicenseInvalidException : PdfViewerException
    {
        public PdfLicenseInvalidException(string message = "") : base(message) { }
    }
    public class PdfPasswordException : PdfViewerException
    {
        public PdfPasswordException(string message = "") : base(message) { }
    }
    public class PdfUnsupportedFeatureException : PdfViewerException
    {
        public PdfUnsupportedFeatureException(string message = "") : base(message) { }
    }

    public class PdfRequestCanceledException : PdfViewerException
    {
        public PdfRequestCanceledException(string message = "") : base(message) { }
    }
}

File: Java/PdfViewerAPI_R2/java/com/pdf_tools/pdfviewer/Model/PdfViewerException.java
package com.pdf_tools.pdfviewer.Model;

import java.util.List;

/**
 * @author fwe
 * Exception which passes information about issues opening and reading a pdf.
 */
/**
 * @author fwe
 *
 */
public class PdfViewerException extends Exception
{

    /**
     * 
     */
    private static final long serialVersionUID = 7972336167506966108L;
    private String message;

    /**
     * Create exception
     * 
     * @param message
     *            cause of exception
     */
    public PdfViewerException(String message)
    {
        this.message = message;
    }

    /*
     * (non-Javadoc)
     * 
     * @see java.lang.Throwable#getMessage()
     */
    @Override
    public String getMessage()
    {
        return message;
    }

    /**
     * @author fwe Exception indicating that an operation could not be executed,
     *         because there is no file open
     */
    public static class NoFileOpenedException extends PdfViewerException
    {
        private static final long serialVersionUID = -7611356292076531076L;

        public NoFileOpenedException()
        {
            super("No file opened to perform this operation");
        }
    }

    /**
     * @author fwe Exception indicating that there was an error for drawing the
     *         image of the pdf file
     */
    public static class GeneralDrawException extends PdfViewerException
    {
        private static final long serialVersionUID = -78416870602744270L;

        public GeneralDrawException(String message)
        {
            super(message);
        }
    }

    /**
     * @author fwe Exception indicating that the file that was requested to be
     *         opened could not be found
     */
    public static class PdfFileNotFoundException extends PdfViewerException
    {
        private static final long serialVersionUID = 8112096691738611681L;

        public PdfFileNotFoundException(String message)
        {
            super(message);
        }
    }

    /**
     * @author fwe Exception indicating that the opened file is password
     *         protected and no valid password has been provided
     */
    public static class PdfPasswordException extends PdfViewerException
    {
        private static final long serialVersionUID = 4334206169115389345L;

        public PdfPasswordException(String message)
        {
            super(message);
        }
    }

    /**
     * @author fwe Exception indicating that there was an internal caching error
     */
    public static class PageNotCachedException extends PdfViewerException
    {
        private static final long serialVersionUID = 661698484905384626L;

        public PageNotCachedException(List<Integer> missingPages, String message)
        {
            super(message);
            this.missingPages = missingPages;
        }

        public List<Integer> missingPages;
    }

    /**
     * @author fwe Exception indicating that the given license key is not valid
     */
    public static class PdfLicenseInvalidException extends PdfViewerException
    {
        private static final long serialVersionUID = 9013387892314346488L;

        public PdfLicenseInvalidException(String message)
        {
            super(message);
        }
    }
    
    /**
     * @author pgl
     * Exception when saving failed due to no file access
     */
    public static class PdfNoFileAccessException extends PdfViewerException
    {
        /**
         * 
         */
        private static final long serialVersionUID = 1L;

        public PdfNoFileAccessException(String message)
        {
            super(message);
        }
    }

    /**
     * @author fwe Exception indicating that the opened file has no valid pdf
     *         header and is thus either corrupt or not a pdf file at all
     */
    public static class PdfFileCorruptException extends PdfViewerException
    {
        private static final long serialVersionUID = -7088022809740319555L;

        public PdfFileCorruptException(String message)
        {
            super(message);
        }
    }

    /**
     * @author fwe Exception indicating that the opened pdf tries to use
     *         features that are not yet supported by the rendering engine
     */
    public static class PdfUnsupportedFeatureException extends PdfViewerException
    {
        private static final long serialVersionUID = 3860519604963591630L;

        public PdfUnsupportedFeatureException(String message)
        {
            super(message);
        }
    }
}
